File: Backend/farmguider/src/main/java/pl/edu/uwm/farmguider/models/crop/dtos/CropResponseDTO.java
package pl.edu.uwm.farmguider.models.crop.dtos;

import lombok.Builder;

import java.math.BigDecimal;
import java.time.LocalDate;

public record CropResponseDTO(Long cropId,
                              String cropType,
                              LocalDate sowingDate,
                              LocalDate harvestDate,
                              LocalDate expectedHarvestStartDate,
                              LocalDate expectedHarvestEndDate,
                              BigDecimal yield,
                              BigDecimal expectedYield) {

    @Builder
    public CropResponseDTO {}

}

File: Frontend/farmguider-app/src/entities/CropResponseDTO.ts
export default interface CropResponseDTO {
    cropId: number;
    cropType: string;
    sowingDate: Date;
    harvestDate: Date | null;
    expectedHarvestStartDate: Date;
    expectedHarvestEndDate: Date;
    yield: number | null;
    expectedYield: number;
}